Port of Shiru's famitone2 utils

R-Gerard opened the pull request:
- nesasmc
- nsf2data
- text2data

clbr commented:
- these tools do not use libpng, and they can probably omit the link lines, relying on the automatic make one
- please include a separate commit for any changes. Ie "Include shiru's tools" which at most does dos2unix, then a second commit for any Linux changes

R-Gerard commented:
Okay, I've split up the changes as per your suggestions. The initial import has the following changes from Shiru's code:
- Ran dos2unix on all files
- Converted cpu2a03.h from iso-8859-1 to utf-8
- Removed trailing whitespace from all files

clbr commented:
Thanks!
